docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+* Provisioning includes attaching the volume to a host.
+* Advise the scheduler with pod placement, keeping the volume rule logic contained to the provider. 
+* Attach disks to hosts (for cloud providers), keeping cloud credentials contained to the provider's host.
+
+Stores added manually via the API would be picked up by a running controller of that type (via a watch).  This allows manual creation and management of storage.  Some controllers will have implementations able to support dynamic creation of resources more easily than others.
+
+
+#### Pod to specify storage requirements
+
+```
+
+	Pod.Spec.Volumes.Volume 
+			Name
+			VolumeSource -- left empty by user, will be bound with available storage from the pool
+			PersistenceRequirements
+				Mode				ReadWriteOnce | ReadOnlyMany
+				RetentionPolicy 	StoreRetentionPolicy (RetainOnDelete|RecycleOnDelete)				Resources			ResourcesList (size, iops, throughput)
+				Selector			Labels to identify the PersistentStore backing implementation
+
+```
+
+`PersistenceRequirements` are the capabilities and performance characteristics a pod author can request and rely on for the entire lifecycle of the volume.  Requests for storage that do not match available storage will go unfulfilled.
+
+
+#### Storage discovery 
+
+An API and CLI is needed to view available and in-use stores by namespace.
+
+**Use Case**:  Susie needs to crunch some data and serve it afterwards.  A single pod to create data is sufficient, but she estimates needing 10 web front ends to serve the data afterwards.  For this use case, the data does not change once created.  Susie first looks to see what is available in the cluster.
+
+```
+$ cluster/kubectl.sh get storage
+
+NAME                LABELS              MODES               RESOURCES
+_pv_vol_01          name=my-data        RWO,ROX             size=5gb,iops=30
+_pv_vol_02                              RWO,ROX             size=5gb,iops=30
+_pv_vol_03          name=other-data     RWO,ROX             size=25gb,iops=30
+
+```
+
+#### Request persistent storage for a pod
+
+Susie is happy to see storage that can be mounted many times (ReadOnlyMany).  That satisfies her load balancer requirement.  Susie creates an pod image with her number crunching code and requests persistence for her volume.
+
+Susie's YAML looks something like this for the pod that creates her data:
+
+
+```yaml
+
+id: creator-of-data
+kind: Pod
+namespace: ns
+apiVersion: v1beta1
+desiredState:
+  manifest:
+    version: v1beta1
+    id: creator-of-data
+    containers:
+      - name: numbercruncher
+        image: ns/makesomdata:v1
+        command:
+          - /run.sh
+        volumeMounts:
+          - name: data
+            mountPath: /some_data
+    volumes:
+      - name: data
+        persistenceRequirements:
+        	availableModes: readwriteonce | readonlymany
+        	currentMode: readwriteonce
+        	retentionPolicy: retainondelete
+        	resources:
+        		- size: 5
```

Comment:
Actually, it does not support units.  It supports scalar suffixes.  Units are implied by the resource type.